### ISSUE TITLE
Allow loading Model, Optimizer, and Scheduler Training Checkpoints from Hugging Face

### ISSUE THREAD

**Issue by ramppdev** [autrainer/autrainer#134]
Right now a model can specify a `model_checkpoint`, `optimizer_checkpoint`, and `scheduler_checkpoint` from a local path.
For `autrainer inference` we support loading models from Hugging Face in combination with a checkpoint.

This functionality could be added to the model, optimizer, and scheduler checkpoints used for training as well, by also allowing the user to pass sth like:
```yaml
# conf/model/SomeModel.yaml
id: SomeModel
...

model_checkpoint: hf:repo_id[@revision][:subdir]#local_dir
# or 
model_checkpoint:
    link: hf:repo_id[@revision][:subdir]#local_dir
    checkpoint: best # best, last, epoch / iteration number, defaults to best
```

**Comment by ramppdev** [comment]
> I think we should also handling an automatic adjustment of the `output_dim` since that is a parameter that might change often when fine-tuning

How is this meant? Users should be able to use checkpoints with arbitrary output dimensions when specifying a `model_checkpoint`?

**Comment by ATriantafyllopoulos** [comment]
> How is this meant? Users should be able to use checkpoints with arbitrary output dimensions when specifying a model_checkpoint?

Exactly, that is the point, isn't it? To streamline fine-tuning within `autrainer`?

**Comment by ramppdev** [comment]
> > How is this meant? Users should be able to use checkpoints with arbitrary output dimensions when specifying a model_checkpoint?
> 
> Exactly, that is the point, isn't it? To streamline fine-tuning within `autrainer`?

Yup and we already support this by default. See the section in the model docs regarding the [model checkpoint](https://autrainer.github.io/autrainer/modules/models.html). `skip_last_layer` is set to true by default and will only load the state up until the last layer such that the architecture specified in `model_checkpoint` only has to match up until the last layer.
